src/NuGet.Clients/NuGet.CommandLine/Common/MSBuildProjectSystem.cs

```diff
@@ -46,17 +48,71 @@ public NuGetFramework TargetFramework
         {
             get
             {
-                string moniker = GetPropertyValue("TargetFrameworkMoniker");
+                string moniker = GetTargetFrameworkString();
+
                 if (String.IsNullOrEmpty(moniker))
                 {
                     return null;
                 }
-                return NuGetFramework.Parse(moniker);
+
+                var framework = NuGetFramework.Parse(moniker);
+
+                // if the framework is .net core 4.5.1 return windows 8.1
+                if (framework.Framework.Equals(FrameworkConstants.FrameworkIdentifiers.NetCore)
+                    && framework.Version.Equals(Version.Parse("4.5.1.0")))
+                {
+                    return new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Windows, Version.Parse("8.1"), framework.Profile);
```

Comment:
currently it does not exist any where in Frameworks, but i can add a new api something like GetSupportedFramework(framwork) in NugetFramework itself so that the same api could be used across vs, commandline,